### DISCUSSION TITLE
Packaging IBM/RexxLA Open Object REXX

### DISCUSSION THREAD

**Discussion by johnsonjh** [Homebrew/discussions#3449] [Tap maintenance and Homebrew development] [answered]
@carlocab (and others ... )

My quality of life would be immensely improved if I could package (ex-IBM) RexxLA [Open Object REXX](https://www.oorexx.org/) for Homebrew.  I have what is almost an acceptable formula ready to go, but need some guidance.

This is the reference Open Object Rexx implementation from IBM, a long time commercial product, and now an open source project managed by the [Rexx Language Association](https://www.rexxla.org/). This project is the official successor and the official recommended upgrade path for all of IBM's commercial Object Rexx customers. This is a mature but [extremely active](https://sourceforge.net/p/oorexx/code-0/HEAD/tree/) and maintained project.

The 'problem' for Homebrew packaging is that the last official release of RexxLA's ooRexx was 4.2 back in 2014. This release is now thoroughly obsolete, doesn't support major features most people are using now like Unicode, and would require major patching to get running on current Homebrew.  With rare exceptions, 100% of users are using the [release 5 SVN versions](https://sourceforge.net/p/oorexx/code-0/HEAD/tree/), especially in production environments; the last released version is essentially a legacy artifact.

There is near-universal consensus that the current code is, in every possible way, more stable, more conforming, more performant, and much more versatile than the last officially tagged release version.  There *have* been some efforts to get a new official release version tagged out (since 2018), but for various reasons this has not occurred. This presents the currently situation where the current "release" version is ill suited for almost all usage and the unreleased [SVN version](https://sourceforge.net/p/oorexx/code-0/HEAD/tree/) is the version that is in wide usage by the vast majority today... The SVN is actively developed but very stable - version 5 SVN is essentially a "stable rolling release".

I'd love to package this, if packaging it would be allowed, with 'releases' made based on tagging a particular SVN commit.  Future maintenance would surely not be a burden on anyone - the build system is thoroughly modern CMake (and it easily compiles on just about *any system* with no effort, *not* only AIX and Linux). It would be extremely unlikely that any major breaking changse would be introduced. There is a comprehensive test framework available that allows for completely automated unit and regression testing.  

I would propose that until an official 5.0 release is made, the test suite could be run, and assuming no regressions or failures, a new Homebrew version could be made on rolling schedule (such as every 3 to 6 months), or as needed (as might be the case if an official 5.0 release version would be tagged.)  This is really is a completely automated process already.

Before I commit to packaging, would this plan be something that the maintainers could find acceptable?  

PS - as a complete and total newbie to Ruby - having never had the need or desire to even look at it before a day ago - I must personally thank you all for your patience and willingness to help me getting new formulas accepted and merged.  I hope that my next efforts will need less hand-holding (and I'm also working out a way to be able to have more direct access to a Macintosh for development, rather than my current round-about setup, which will help immensely).

**Comment by SMillerDev** [comment, marked as answer]
> I'd love to package this, if packaging it would be allowed, with 'releases' made based on tagging a particular SVN commit.

I'm going to pass on this for homebrew-core. It makes the "when do we update" impossible to determine.

You can always build it in a tap though.

**Reply by johnsonjh** [reply]
Understood - if the situation there changes (I'm hoping for a real release and then something like a "stable" branch) I'll revisit it up again. I'll probably go the tap route regardless. Thanks
